src/v2i-hub/CARMACloudPlugin/src/CARMACloudPlugin.cpp

```diff
@@ -548,25 +588,47 @@ void CARMACloudPlugin::ConvertString2Pair(std::pair<string,string> &str_pair, co
 	str_pair = std::make_pair(key, value);
 }
 
+QByteArray CARMACloudPlugin::UncompressBytes(const QByteArray compressedBytes) const
+{
+    z_stream strm;
+    strm.zalloc = nullptr;
+    strm.zfree = nullptr;
+    strm.opaque = nullptr;
+    strm.avail_in = compressedBytes.size();
+    strm.next_in = (Byte *)compressedBytes.data();

```

Comment:
Is there any inline comments you can leave here to clarify what you are doing and what function/method calls you are using to accomplish this? What does inflateInit2 do?  Why is the buffer size 4092? What is MAX_WBits? What is Z_FINISH? Lastly is all TCM data sent as compressed data now regardless if  list_tcm is set to true or false?

Reply:
If it is set to false, it will return plain data. Only true, it will compress the data.